packages/lesswrong/server/callbacks/votingCallbacks.ts

```diff
@@ -15,14 +26,16 @@ const collectionsThatAffectKarma = ["Posts", "Comments", "Revisions"]
 voteCallbacks.castVoteAsync.add(function updateKarma({newDocument, vote}: VoteDocTuple, collection: CollectionBase<DbVoteableType>, user: DbUser) {
   // only update karma is the operation isn't done by the item's author
   if (newDocument.userId !== vote.userId && collectionsThatAffectKarma.includes(vote.collectionName)) {
-    void Users.rawUpdateOne({_id: newDocument.userId}, {$inc: {"karma": vote.power}});
+    const owners = getDocumentOwners({newDocument, vote});
+    void Users.rawUpdateMany({_id: {$in: owners}}, {$inc: {karma: vote.power}});
   }
 });
 
 voteCallbacks.cancelAsync.add(function cancelVoteKarma({newDocument, vote}: VoteDocTuple, collection: CollectionBase<DbVoteableType>, user: DbUser) {
   // only update karma is the operation isn't done by the item's author
   if (newDocument.userId !== vote.userId && collectionsThatAffectKarma.includes(vote.collectionName)) {
-    void Users.rawUpdateOne({_id: newDocument.userId}, {$inc: {"karma": -vote.power}});
+    const owners = getDocumentOwners({newDocument, vote});
```

Comment:
With this implementation, I think, if a post is upvoted, then a co-author is added, and then someone removes their upvote, the co-author will end up with net-negative karma.

Co-authors being added after initial publication is not uncommon in my experience since often people publish before getting our help on adding coauthors.

Reply:
Fixed!